presto-postgresql/src/main/java/com/facebook/presto/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -105,17 +106,24 @@ protected ResultSet getTables(Connection connection, Optional<String> schemaName
     }
 
     @Override
-    protected String toSqlType(Type type)
+    public WriteMapping toWriteMapping(Type type)
     {
         if (VARBINARY.equals(type)) {
-            return "bytea";
+            return sliceMapping("bytea", (SliceWriteFunction) varbinaryColumnMapping().getWriteFunction());
         }
-
-        return super.toSqlType(type);
+        else if (type.getTypeSignature().getBase().equals(StandardTypes.JSON)) {
+            return sliceMapping("jsonb", ((statement, index, value) -> {
+                PGobject pgObject = new PGobject();
+                pgObject.setType("json");
+                pgObject.setValue(value.toStringUtf8());
+                statement.setObject(index, pgObject);
+            }));
+        }
+        return super.toWriteMapping(type);
```

Comment:
It looks like we're also adding support for JSON types here? If so, can the JSON support be in a separate commit?